pkg/services/sqlstore/permissions/dashboard_test.go

```diff
@@ -160,6 +161,137 @@ func TestIntegration_DashboardPermissionFilter(t *testing.T) {
 	}
 }
 
+func TestIntegration_DashboardPermissionFilter_WithSelfContainedPermissions(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	type testCase struct {
+		desc                    string
+		queryType               string
+		permission              dashboards.PermissionType
+		signedInUserPermissions []accesscontrol.Permission
+		expectedResult          int
+	}
+
+	tests := []testCase{
```

Comment:
nit: could you add a test where the user is not supposed to access any dashboards and folders?

Reply:
Done, thanks!